mne/io/base.py

```diff
@@ -363,6 +368,31 @@ def __init__(self, info, preload=False,
         self._comp = None
         self._filenames = list(filenames)
         self.orig_format = orig_format
+        # Sanity check and set original units, if provided by the reader:
+        if orig_units:
+            if not isinstance(orig_units, dict):
+                raise ValueError('orig_units must be of type dict, but got '
+                                 ' {}'.format(type(orig_units)))
+
+            # original units need to be truncated to 15 chars, which is what
+            # the MNE IO procedure also does with the other channels
+            orig_units_trunc = [ch[:15] for ch in orig_units]
+
+            # STI 014 channel is native only to fif ... for all other formats
+            # this was artificially added by the IO procedure, so remove it
+            ch_names = list(info['ch_names'])
+            if ('STI 014' in ch_names) and not \
+               (self.filenames[0].endswith('.fif')):
+                ch_names.remove('STI 014')
+
+            # Each channel in the data must have a corresponding channel in
+            # the original units.
+            ch_correspond = [ch in orig_units_trunc for ch in ch_names]
+            if not all(ch_correspond):
+                ch_without_orig_unit = ch_names[ch_correspond.index(False)]
+                raise ValueError('Channel {0} has no associated original '
+                                 'unit.'.format(ch_without_orig_unit))
+            self._orig_units = orig_units
```

Comment:
This should probably not be indented. That way `raw._orig_units` is always there, but could be `None`. This is easier to work with using `if raw._orig_units is not None` than `if hasattr(raw, '_orig_units')` or a `try/except`